src/app/clusters/webrtc-transport-requestor-server/webrtc-transport-requestor-server.cpp

```diff
@@ -0,0 +1,431 @@
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleOffer(ctx_, req); });
+        break;
+    case Commands::Answer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::Answer::DecodableType>(
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleAnswer(ctx_, req); });
+        break;
+    case Commands::ICECandidates::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ICECandidates::DecodableType>(
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleICECandidates(ctx_, req); });
+        break;
+    case Commands::End::Id:
+        CommandHandlerInterface::HandleCommand<Commands::End::DecodableType>(
+            ctx, [this](HandlerContext & ctx_, const auto & req) { HandleEnd(ctx_, req); });
+        break;
+
+    default:
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::UnsupportedCommand);
+        break;
+    }
+}
+
+uint16_t WebRTCTransportRequestorServer::GenerateSessionId()
+{
+    static uint16_t lastSessionId = 0;
+    uint16_t candidateId          = 0;
+
+    while (true)
+    {
+        candidateId = lastSessionId++;
+
+        // Handle wrap-around per spec
+        if (lastSessionId > kMaxSessionId)
+        {
+            lastSessionId = 0;
+        }
+
+        if (FindSession(candidateId) == nullptr)
+        {
+            break;
+        }
+    }
+
+    return candidateId;
+}
+
+bool WebRTCTransportRequestorServer::IsPeerNodeSessionValid(uint16_t sessionId, HandlerContext & ctx)
+{
+    NodeId peerNodeId           = GetNodeIdFromCtx(ctx.mCommandHandler);
+    FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    // Check if the session ID is in the existing sessions list
+    WebRTCSessionTypeStruct * existingSession = FindSession(sessionId);
+
+    if (!existingSession)
+    {
+        return false;
+    }
+
+    // Also check that the existing session belongs to the same PeerNodeID / Fabric
+    // If it doesn’t match, return false
+    if (peerNodeId != existingSession->peerNodeID || peerFabricIndex != existingSession->GetFabricIndex())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+void WebRTCTransportRequestorServer::HandleOffer(HandlerContext & ctx, const Commands::Offer::DecodableType & req)
+{
+    // Extract command fields from the request
+    uint16_t sessionId = req.webRTCSessionID;
+
+    NodeId peerNodeId           = GetNodeIdFromCtx(ctx.mCommandHandler);
+    FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    // Check if the session ID is in the existing sessions list
+    WebRTCSessionTypeStruct * existingSession = FindSession(sessionId);
+    // Create arguments for WebRTCTransportRequestorDelegate.
+    WebRTCTransportRequestorDelegate::OfferArgs args;
+    WebRTCSessionTypeStruct outSession;
+
+    if (!existingSession)
+    {
+        // As per specification, sessionId represents the established session
```

Comment:
I don't understand this comment or the logic here.  Spec says:

> This command SHALL respond with a response status of NOT_FOUND, if the WebRTCSessionID is not valid

If we don't have`existingSession`, is it "valid"?  Or is that something the delegate needs to sort out?  Or is this what the TODO right below this is about and we actually need an issue (to be linked in the TODO) to make this code match the spec?

Reply:
Yes, I think the Offer is received here after the SolicitOfferResponse, which should have created the session. So, not finding the session should be an exception path.